rmw_zenoh_cpp/src/detail/zenoh_utils.hpp

```diff
@@ -15,37 +15,36 @@
 #ifndef DETAIL__ZENOH_UTILS_HPP_
 #define DETAIL__ZENOH_UTILS_HPP_
 
-#include <zenoh.h>
+#include <zenoh.hxx>
 
 #include <chrono>
 #include <functional>
 #include <optional>
+#include <vector>
 
 #include "rmw/types.h"
 
 namespace rmw_zenoh_cpp
 {
 ///=============================================================================
-void
-create_map_and_set_sequence_num(
-  z_owned_bytes_t * out_bytes, int64_t sequence_number,
-  uint8_t gid[RMW_GID_STORAGE_SIZE]);
+zenoh::Bytes create_map_and_set_sequence_num(
+  int64_t sequence_number, std::vector<uint8_t> gid);
 
 ///=============================================================================
 // A class to store the replies to service requests.
 class ZenohReply final
 {
 public:
-  ZenohReply(const z_loaned_reply_t * reply, std::chrono::nanoseconds::rep received_timestamp);
+  ZenohReply(const zenoh::Reply & reply, std::chrono::nanoseconds::rep received_timestamp);
```

Comment:
Instead of cloning the reply, why don't we std:move a reply into a `ZenohReply` object?

Reply:
Yes, I think that would be more efficient if that works.

Reply:
Actually I think this is fine now.  As a constref, this is basically passing the reply in as a pointer, and then in the implementation of the `ZenohReply` constructor we are doing the clone.  So I think this is only one copy, which we need to do either way.